pkg/jaeger/store/get_operations.go

```diff
@@ -76,10 +78,13 @@ func getOperations(ctx context.Context, conn pgxconn.PgxConn, query spanstore.Op
 	return operationsResp, nil
 }
 
-func textArraytoStringArr(s pgtype.TextArray) ([]string, error) {
-	var d []string
-	if err := s.AssignTo(&d); err != nil {
-		return []string{}, fmt.Errorf("assign to: %w", err)
+func textArraytoStringArr(s pgtype.FlatArray[pgtype.Text]) ([]string, error) {
+	d := make([]string, len(s))
+	for i, v := range s {
+		if !v.Valid {
+			return nil, errors.New("can't assign NULL to string")
```

Comment:
> I wonder if we should go with string pointers? Or maybe just use empty string for NULL? I feel that erroring is maybe too strick. For example if some array values have NULLs (legacy?) it will actually break the connector.

The behaviour didn't change, the `s.AssignTo` method will return an error if there is a NULL value, because it will try to assign a nil to a string. I modify a test from the deprecated pgtype package to check this:

```go
func TestTextArrayDecodeTextNull(t *testing.T) {
	textArray := &pgtype.TextArray{}
	err := textArray.DecodeText(nil, []byte(`{abc,NULL}`))
	require.NoError(t, err)
	require.Len(t, textArray.Elements, 2)
	assert.Equal(t, pgtype.Present, textArray.Elements[0].Status)
	assert.Equal(t, pgtype.Null, textArray.Elements[1].Status)
	var a []string
	err = textArray.AssignTo(&a)
	require.Errorf(t, err, "cannot assign NULL to *string")
}
```

Reply:
> Btw I've also noted that we have `TextArrayToSlice` function.

Yeah, both methods are almost the same. The thing is that `TextArrayToSlice` is basically replacing the [Get method from pgutf8str](https://github.com/timescale/promscale/pull/1809/files#diff-537f9ac4eefdd4668983d9efa6d91797a67abac9d26259193a155d1f94e2e80bL46-L52)  (The code on the left). It does the `revertSanitization` and I just realised that I left a question there, because the method panics, the old one also has a panic, but maybe we can handle it as an error instead.